node/consensus/data/data_clock_consensus_engine.go

```diff
@@ -338,6 +340,39 @@ func (e *DataClockConsensusEngine) Start() <-chan error {
 
 	e.state = consensus.EngineStateCollecting
 
+	go func() {
```

Comment:
Are there other places where this could've been placed in ? Right now it affects only the data clock consensus engine, but maybe it should check for stalls in all engines long term ?